File: mobile/src/types/license.ts
export type LicenseStatus = 'ACTIVE' | 'SUSPENDED';

export interface License {
  id: string;
  license_no: string;
  qr_token: string;
  status: LicenseStatus;
  points: number;
  issued_at: string;
  expiry_at: string;
}

File: backend/app/schemas/license.py
import uuid

from pydantic import BaseModel

from app.models.license import LicenseStatus
from app.schemas.common import UtcDateTime


class LicenseRead(BaseModel):
    id: uuid.UUID
    license_no: str
    qr_token: str
    status: LicenseStatus
    points: int
    issued_at: UtcDateTime
    expiry_at: UtcDateTime

    model_config = {"from_attributes": True}
